apps/design/backend/src/app.ts

```diff
@@ -304,52 +307,84 @@ export function buildApi(ctx: AppContext) {
 
     async loadElection(
       input: {
-        electionData: string;
+        upload: ElectionUpload;
         newId: ElectionId;
         orgId: string;
       },
       context: ApiContext
     ): Promise<Result<ElectionId, Error>> {
-      const parseResult = safeParseElection(input.electionData);
-      if (parseResult.isErr()) return parseResult;
-      const sourceElection = parseResult.ok();
-      const { districts, precincts, parties, contests } =
-        regenerateElectionIds(sourceElection);
-      // Split candidate names into first, middle, and last names, if they are
-      // not already split
-      const contestsWithSplitCandidateNames = contests.map((contest) => {
-        if (contest.type !== 'candidate') return contest;
-        return {
-          ...contest,
-          candidates: contest.candidates.map((candidate) => {
-            if (
-              candidate.firstName !== undefined &&
-              candidate.middleName !== undefined &&
-              candidate.lastName !== undefined
-            ) {
-              return candidate;
-            }
-            return {
-              ...candidate,
-              ...splitCandidateName(candidate.name),
+      const electionResult = ((): Result<Election, Error> => {
+        switch (input.upload.format) {
```

Comment:
The reused common code paths in this method are so minimal that I think could be clearer to have explicit methods like `loadElectionVxf()`, `loadElectionMsSems()`, etc (or something along those lines), since we know which path we're taking at request-time on the client anyway. Wary of the potential road to large switch-case method with additional special-case fields in the input. WDYT?

Reply:
I'll mock up the change and see how it feels

Reply:
Ok, here's a sketch of what it looks like: https://github.com/votingworks/vxsuite/pull/7405/files. I think the server side does end up a bit clearer, but the frontend is a bit hairy, mainly because of way the form is weirdly split across two parts of a modal. The selector for which file format to use is part of the modal contents, but the action buttons are a separate prop to the modal. I would like to just separate out the forms into two separate components with separate form state, but I can't think of a simple way to do that. I could swap out the entire modal, but then both modals would need to have the format selector within them. Curious if you have any ideas here or any new thoughts after seeing what this code looks like